### ISSUE TITLE
Extract Embedded Alt Text To Alt Text Metadata Field On Fileset and Display on Work Showpage

### ISSUE THREAD

**Issue by KevinJonesMeta** [OregonDigital/OD2#3637]
### Descriptive summary

Alt text values (primarily on image files) will be embedded on files to support WCAG 2.1 compliance. Oregon Digital needs to extract that embedded metadata and populate an alt text field on fileset to display in file detail section of Works and Files on showpage. this value will also get added to IIIF manifest to travel with the work if the manifest is used elsewhere (see related ticket).

Metadeities and other submitters will populate both the IPTC Alt Text field and the ImageDescription field on files to ensure technical compliance and functional utility of the alt text value. ImageDescription appears to be an XMP field whose value is duplicated in the IPTC Description field. See p120_3375_bd.tif for an example. 

Metadeities has created a predicate to hold alt text values on the fileset:

http://opaquenamespace.org/ns/altText

Example files for reference (shared with @CGillen in box):

p120_3375_bd.tif - This file has a value in the IPTC Alt Text field and XMP ImageDescription fields on the file. The ImageDescription field value also populates the IPTC Description field on the file. the ImageDescription tends to be more readily accessible to image display apps but it is not an Alt Text field per se.

p120_3376.tif - This file has a value in the IPTC Alt Text field. Though using the correct field according to the IPTC standard, it does not display in many image display apps.

### Expected behavior

A file that is imported via Bulkrax or the UI form that contains embedded alt text metadata in the ImageDescription field has that data extracted and it populates the alt text field on the fileset. That fileset field displays on the showpage in the same area as the accessibilityFeature values for each fileset.

### Related work

#3636 

### Accessibility Concerns

Add any information here to indicate any known or suspected accessibility issues for this ticket

**Comment by lamtu1** [comment]
@KevinJonesMeta I was wondering if you could link up to the work with `p120_3375_bd.tif ` and `p120_3376.tif` so I can see and understand the metadata and the content of it while I am trying to plan out for it? Thank You!

@briesenberg07 Thank you for the new predicate and I was also wondering if the MAP will be updated for this predicate in our drive by any chance? Thank you

**Comment by KevinJonesMeta** [comment]
@lamtu1 I shared the files with you in Box. Is there something else that would help?

**Comment by lamtu1** [comment]
> [@lamtu1](https://github.com/lamtu1) I shared the files with you in Box. Is there something else that would help?

I think this will be good to know what to as I am start to implement locally, and for now i think that its but if anything, I will reach out more.

**Comment by briesenberg07** [comment]
@lamtu1 here is the predicate URI to use when configuring the file set field...

http://opaquenamespace.org/ns/altText

...and yes, metadata team will be sure to document this in MAP. Is the MAP info needed for your work? 

@KevinJonesMeta let's work up MAP details? I'm realizing I may have questions about file set fields... for example, not sure whether file set fields can be configured as repeatable (even though, pretty sure that we do _not_ want this one to be.)

**Comment by lamtu1** [comment]
@briesenberg07 Yes the MAP is important for me to setup the predicate into the FileSet, but beside that I can always start out with the basic layout from another predicate we did before and modify this new predicate MAP once we have it updated.

**Comment by KevinJonesMeta** [comment]
(for tracking) Metadeities discussed in Slack and agreed that having extracted alt text viewable on the fileset metadata page in `Works and Files=>Fileset title link` is good in addition to having it under `accessibilityFeature` values for the file on work showpage. Will not be edited in form, only when file is handled outside Oregon Digital.

**Comment by lamtu1** [comment]
@KevinJonesMeta  For QA: Try to create a work and upload an image (with & without) alt_text through the UI form and see if the alt_text can be extracted.

**Comment by KevinJonesMeta** [comment]
QA Pass in UI form. 

<img width="1446" height="867" alt="Image" src="https://github.com/user-attachments/assets/e88a39cf-8a73-4c8b-b0bd-109873133b55" />

@lamtu1 Can I test via Bulkrax as well or should I wait?

**Comment by lamtu1** [comment]
@KevinJonesMeta You can QA for Bulkrax as well

**Comment by KevinJonesMeta** [comment]
QA fail for Bulkrax. See https://staging.oregondigital.org/importers/274

Error: ActiveFedora::UnknownAttributeError

**Comment by lamtu1** [comment]
Thank you for testing that, and will get on with bulkrax fixing for this

**Comment by KevinJonesMeta** [comment]
QA Pass. Test batch in Bulkrax completed, alternative text field label is correct, alt text value appeared in Works and Files under Accessibility Feature as expected.

**Comment by KevinJonesMeta** [comment]
@lamtu1 It looks like the sample batch that I used has the alternative text in the right place but the viewer doesn't show up. I'll run a new test to confirm. Could that be related to the manifest work or would it be a separate ticket?

**Comment by KevinJonesMeta** [comment]
@lamtu1  the second test looks right and shows in viewer. Not sure what happened. If it comes up again I'll put in a new ticket.

**Comment by lamtu1** [comment]
@KevinJonesMeta Yes this is correct, this is part of the ticket that I am working on that deal with the manifest and the display of it in universal viewer, which the display of it will come from that ticket. But it is being wonky or something, feel free to draft another one up and I will take a look at it.
